CONTRIBUTING.md

```diff
@@ -14,26 +14,93 @@ If in doubt about a policy, please [ask on our Slack](https://communityinviter.c
+Schemas are only changed _after_ the specification is changed.
+Changes are made to the YAML versions on the `main` branch.
+The JSON versions are generated when publised to the spec site, at which time the `WORK-IN-PROGRESS` URI placeholders are replaced with the publication date.
+
+### Authoritative source of truth
+
+The [spec site](https://spec.openapis.org) is the source of truth.
+
+This changed in 2024, as the markdown files on `main` do not include certain credits and citations.
+
+## Development and publication process
+
+As of October 2024 (post-OAS 3.0.4 and 3.1.1), the OAS is developed in the `src/oas.md` file on minor release `X.Y-dev` branches that are derived from the baseline `dev` branch.
+
+All work **MUST be done on a fork**, using a branch from the _earliest relevant and [active](#active-branches)_ `X.Y-dev` branch, and then submitted as a PR to that `X.Y-dev` branch.
+For example, if a change in November 2024 apples to both 3.1 and 3.2, the PR would go to the `v3.1-dev` branch, which will be merged up to `v3.2-dev` before the next 3.2 release.
+
+Releases are published to the [spec site](https://spec.opanapis.org) by creating an `X.Y.Z-rel` branch where `src/oas.md` is renamed to the appropriate `versions/X.Y.Z.md` file and them merged to `main`.  The HTML versions of the OAS are automatically generated from the `versions` directory on `main`.  This renaming on the `X.Y.Z-rel` branch preserves the commit history for the published file on `main` when using `git log --follow` (as is the case for all older published files).
+
+For information on the branch and release strategy for OAS 3.0.4 and 3.1.1 and earlier, see the comments in [issue #3677](https://github.com/OAI/OpenAPI-Specification/issues/3677).
+
+### Branch diagram (3.1.2, 3.2.0, and later)
+
+```mermaid
+gitGraph TB:
+  commit id:"merge 3.1.1.md to main" tag:"3.1.1"
+  branch dev order:1
+  commit id:"rename 3.1.1.md to src/oas.md"
+  branch v3.1-dev order:2
+  commit id:"update version in src/oas.md to 3.1.2"
+  checkout dev
+  branch v3.2-dev order:5
+  commit id:"update version in src/oas.md to 3.2.0"
+  commit id:"some 3.2.0 work"
+  checkout v3.1-dev
+  commit id:"a 3.1.x fix"
+  branch v3.1.2-rel order:3
+  commit id:"rename src/oas.md to versions/3.1.2.md"
+  checkout main
+  merge v3.1.2-rel tag:"3.1.2"
+  checkout v3.2-dev
+  commit id:"more 3.2.0 work"
+  checkout v3.1-dev
+  commit id:"update version in src/oas.md to 3.1.3"
+  commit id:"another 3.1.x fix"
+  checkout v3.2-dev
+  commit id:"still more 3.2.0 work"
+  merge v3.1-dev id:"merge 3.1.x fixes before releasing"
+  checkout v3.1-dev
+  branch v3.1.3-rel order:4
+  commit id:"rename src/oas.md to versions/3.1.3.md"
+  checkout v3.2-dev
+  branch v3.2.0-rel order:6
+  commit id:"rename src/oas.md to versions/3.2.0.md"
+  checkout main
+  merge v3.1.3-rel tag:"3.1.3"
+  merge v3.2.0-rel tag:"3.2.0"
```

Comment:
This seems to assume that 3.1.3 and 3.2.0 are published at the same time.

Shouldn't it rather be
```suggestion
  merge v3.1.3-rel tag:"3.1.3"
  checkout dev
  merge v3.1-dev id:"update dev with patch release"
  checkout main
  merge v3.2.0-rel tag:"3.2.0"
```

Reply:
@ralfhandl 

> This seems to assume that 3.1.3 and 3.2.0 are published at the same time.

The point of the example here is to show 3.1.3 and 3.2.0 going out at the same time, as we are likely to do a patch release with each minor release, just like we're doing paired 3.0.4 and 3.1.1 releases.

> Shouldn't it rather be

No, because patch releases don't get merged back to `dev` because they are interleaved across minor release branches.  You can't merge them all consistently.  But we (currently) only ever have one minor release line at a time, so we can merge each minor release down _so it can be the base for the next `X.Y-dev` branch_.  This lets all `X.Y-dev` branches come off of main.  We don't need this for each patch release because they don't get new `-dev` branches, only `-rel` branches which always come off of `dev`.  I'll try to clarify this.